testutil/directory.go

```diff
@@ -143,3 +146,41 @@ func DirSize(path string) (int64, error) {
 	})
 	return size, err
 }
+
+// DirHash returns a hash of all files attribites and their content within a directory.
+func DirHash(path string) ([]byte, error) {
+	hash := md5.New()
+	err := filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() {
```

Comment:
You can continue early and safe one indent for long body below

Reply:
yep , updated